File: CS50/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

//Spilts the string into individual letters and counts them

//Counts the number of words in the sentence

//Counts the number of sentences, anything the ends with "." "!" or "?"

//Index calculation: index = 0.0588 * L - 0.296 * S - 15.8

// L = Average number of letters per 100 words
// S = Average number of sentences per 100 words
float Index(float se, float w, float le);

int main(void)
{
    //Gets the user input sentence
    string text = get_string("Text: ");
    float spaces = 0;
    float l = 0;
    float sentences = 0;
    //Counts the number of spaces and sentence enders in the string
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == ' ')
        {
            spaces++;
        }
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z'))
        {
            l++;
        }
    }
    //float l = strlen(text) - spaces - sentences - special;
    float words = spaces + 1;//Needs to be 1 more as there is no space at the end of the sentence
    if (round(Index(sentences, words, l)) < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (round(Index(sentences, words, l)) > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(Index(sentences, words, l)));
    }
}

//Function for finding the index of the sentences
float Index(float se, float w, float le)
{
    float L = (le / w) * 100;
    float S = (se / w) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    return index;
}

File: CS50/pset6/readability/readability.py
from cs50 import get_string
from sys import exit
import re

# Spilts the string into individual letters and counts them

# Counts the number of words in the sentence

# Counts the number of sentences, anything the ends with "." "!" or "?"

# Index calculation: index = 0.0588 * L - 0.296 * S - 15.8

# L = Average number of letters per 100 words
# S = Average number of sentences per 100 words

# Calculates the index and returns the rounded value


def Index(se, w, le):
    L = (le / w) * 100
    S = (se / w) * 100
    index = (0.0588 * L) - (0.296 * S) - 15.8
    return round(index)


def Count(text):
    l = 0
    # Loops throung the text counting letters
    for i in range(len(text)):
        if text[i].isalpha():
            l += 1
    # Counts the occurance of sentence enders and the words in the string
    sentences = text.count(".") + text.count("?") + text.count("!")
    words = len(re.findall(r'\w+', text)) - text.count("'") - text.count("-")
    return Index(sentences, words, l)

# Gets the text and prints out the grade


def Main():
    text = get_string("Text: ")
    i = Count(text)
    if i > 16:
        print("Grade 16+")
    elif (i < 1):
        print("Before Grade 1")
    else:
        print("Grade " + str(i))


Main()